src/stores/tournament.store.ts

```diff
@@ -13,79 +13,200 @@ const groupBy = <T extends object, K extends keyof T>(items: T[], key: K) => ite
 }), {} as Record<T[K], T[]>);
 
 export const useTournamentStore = defineStore('tournament', () => {
-  const events = ref<Record<number, Event[]>>({});
-  const event = ref<Event[]>([]);
+  const events = ref<Record<number, Event>>({});
   const tournaments = ref<Record<number, Tournament>>({});
-  const tournament = ref<Tournament>();
 
-  async function fetchEvents() {
+  async function fetchAllEvents() {
     try {
       const res = await axios.get<Event[]>('/tournament/event');
-      const ev = res.data;
-      events.value = groupBy(ev, 'year');
+      res.data.forEach((ev) => { events.value[ev.id] = ev; });
       // console.log(years)
-    } catch (error) {
+    } catch (err: any) {
       // console.log(error);
     }
   }
 
+  async function fetchEventsByYear(year: number) {
+    try {
+      const res = await axios.get<Event[]>(`tournament/event/year/${year}`);
+      res.data.forEach((ev) => { events.value[ev.id] = ev; });
+    } catch (err: any) {
+      /* empty */
+    }
+  }
+
+  async function fetchEventsByYears(years: number[]) {
+    await Promise.all<void>(years.map<Promise<void>>((year) => fetchEventsByYear(year)));
```

Comment:
i dont think you should type void here

Reply:
you're right, types can be drop in this case